pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/utils/TopicSubscriptionsVisualizerHtmlUtil.java

```diff
@@ -0,0 +1,143 @@
+                }
+                currentCount = 1;
+                currentStatus = entryStatus;
+            }
+        }
+        if (currentCount > 0) {
+
+            if (currentStatus == TopicSubscriptionsVisualizer.EntryStatus.ACKED ||
+                    currentStatus == TopicSubscriptionsVisualizer.EntryStatus.INDIVIDUALLLY_DELETED) {
+                String cssClass;
+                if (currentStatus == TopicSubscriptionsVisualizer.EntryStatus.ACKED) {
+                    entriesStatsByStatus.totalAcked += currentCount;
+                    cssClass = "acked";
+                } else {
+                    entriesStatsByStatus.totalIndividuallyDeleted += currentCount;
+                    cssClass = "idel";
+                }
+                builder.append("<div class=\"segment-" + cssClass + "\" style=\"width: ");
+                builder.append(singleEntryPercent * currentCount);
+                builder.append("%;\"></div>");
+            } else {
+                entriesStatsByStatus.totalAhead += currentCount;
+                builder.append("<div class=\"segment-ahead\" style=\"width: ");
+                builder.append(singleEntryPercent * currentCount);
+                builder.append("%;\"></div>");
+            }
+
+        }
+        builder.append("</div>");
+        return builder.toString();
+    }
+
+    public static String genLedgerSegment(ManagedLedgerInternalStats.LedgerInfo ledger,
+                                          long totalEntries,
+                                          TopicSubscriptionsVisualizer.LedgerSegmentProgress ledgerSegmentProgress,
+                                          boolean addFirstClass) {
+        String tooltip = genLedgerInfoDescription(ledger);
+        double percentWidth = (double) 100 / totalEntries * ledger.entries;
+        ledgerSegmentProgress.ledgersMapping.put(ledger.ledgerId, new TopicSubscriptionsVisualizer.EntryWidthMapping());
+        for (long i = 0; i < ledger.entries; i++) {
+            ledgerSegmentProgress.ledgersMapping.get(ledger.ledgerId)
+                    .put(i, ledgerSegmentProgress.currentWidth++);
+        }
+
+        return "<div class=\"ledger"
+                + (addFirstClass ? " first": "")
+                + " tooltip\" style=\"width: "
+                + percentWidth + "%\">"
+                + "<span class=\"tooltiptext\">"
+                + tooltip
+                + "</span></div>";
+    }
+
+
+    public static String genLedgerInfoDescription(ManagedLedgerInternalStats.LedgerInfo ledger) {
+        return genDetailsDescription(MAPPER.convertValue(ledger, Map.class));
+
+    }
+
+    public static String genDetailsDescription(Map<String, Object> details) {
+        StringBuilder builder = new StringBuilder("<div class=\"description-details\">");
+        details.forEach((k, v) -> {
+            builder.append("<div class=\"detail-item\"><span class=\"detail-item-key\">");
+            builder.append(k);
```

Comment:
please take into account XSS problems, we have to sanitise all the String coming from the input